### PR TITLE
Code Quality: Replaced icon height & size constants with enums

### PR DESCRIPTION
**Pull request by 0x5bfa**
## Summary

Planned internally.

## PR Checklist

- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [ ] Are there any other steps that were used to validate these changes?
   _None_

## Screenshots

_None_

### CODE REVIEW

File: src/Files.Core/Data/Enums/LayoutColumnsViewIconHeightKind.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) 2023 Files Community
+// Licensed under the MIT License. See the LICENSE.
+
+namespace Files.Core.Data.Enums
+{
+	public enum LayoutColumnsViewIconHeightKind
```
Comment (yaira2) [inline src/Files.Core/Data/Enums/LayoutColumnsViewIconHeightKind.cs:6]:
```suggestion
	public enum ColumnsViewSizeKind
```

File: src/Files.Core/Data/Enums/LayoutColumnsViewIconHeightKind.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+	{
+		Minimum = 28,
+		Small = 32,
+		Regular = 36,
+		Maximum = 40,
```
Comment (yaira2) [inline src/Files.Core/Data/Enums/LayoutColumnsViewIconHeightKind.cs:11]:
```suggestion
		Compact = 0,
		Small = 1,
		Large = 3,
		ExtraLarge = 4,
```

File: src/Files.Core/Data/Enums/LayoutDetailsViewIconHeightKind.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) 2023 Files Community
+// Licensed under the MIT License. See the LICENSE.
+
+namespace Files.Core.Data.Enums
+{
+	public enum LayoutDetailsViewIconHeightKind
```
Comment (yaira2) [inline src/Files.Core/Data/Enums/LayoutDetailsViewIconHeightKind.cs:6]:
```suggestion
	public enum DetailsViewSizeKind
```

File: src/Files.Core/Data/Enums/LayoutDetailsViewIconHeightKind.cs [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+	{
+		Minimum = 28,
+		Small = 32,
+		Regular = 36,
+		Maximum = 40,
```
Comment (yaira2) [inline src/Files.Core/Data/Enums/LayoutDetailsViewIconHeightKind.cs:11]:
```suggestion
		Compact = 0,
		Small = 1,
		Large = 3,
		ExtraLarge = 4,
```

File: src/Files.Core/Data/Enums/LayoutGridViewIconHeightKind.cs [hunk 5]
```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) 2023 Files Community
+// Licensed under the MIT License. See the LICENSE.
+
+namespace Files.Core.Data.Enums
+{
+	public enum LayoutGridViewIconHeightKind
```
Comment (yaira2) [inline src/Files.Core/Data/Enums/LayoutGridViewIconHeightKind.cs:6]:
```suggestion
	public enum GridViewSizeKind
```
This should have values from 0 - 12 but I'm not sure what to name them.

File: src/Files.Core/Data/Enums/LayoutListViewIconHeightKind.cs [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+// Licensed under the MIT License. See the LICENSE.
+
+namespace Files.Core.Data.Enums
+{
+	public enum LayoutListViewIconHeightKind
+	{
+		Minimum = 24,
+		Small = 28,
+		Regular = 32,
+		Maximum = 36,
```
Comment (yaira2) [inline src/Files.Core/Data/Enums/LayoutListViewIconHeightKind.cs:11]:
```suggestion
		Compact = 0,
		Small = 1,
		Large = 3,
		ExtraLarge = 4,
```

File: src/Files.App/Services/Settings/LayoutSettingsService.cs [hunk 7]
```diff
@@ -13,32 +13,32 @@ public LayoutSettingsService(ISettingsSharingContext settingsSharingContext)
 
 		public int ItemSizeDetailsView
```
Comment (yaira2) [inline src/Files.App/Services/Settings/LayoutSettingsService.cs:14]:
```suggestion
		public DetailsViewSizeKind ItemSizeDetailsView
```

File: src/Files.Core/Data/Enums/LayoutIconHeightIncrementKind.cs [hunk 8]
```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) 2023 Files Community
+// Licensed under the MIT License. See the LICENSE.
+
+namespace Files.Core.Data.Enums
+{
+	public enum LayoutIconHeightIncrementKind
```
Comment (yaira2) [inline src/Files.Core/Data/Enums/LayoutIconHeightIncrementKind.cs:6]: The increment enums aren't needed anymore

File: src/Files.Core/Data/Enums/LayoutItemIconSizeKind.cs
Comment (yaira2) [inline src/Files.Core/Data/Enums/LayoutItemIconSizeKind.cs:1]: Let's reverse this for now as it's out of scope for this PR

**Review by yaira2 — APPROVED** [review]
LGTM

File: src/Files.Core/Data/Enums/GridViewSizeKind.cs [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+		Medium = 2,
+		Large = 3,
+		ExtraLarge = 4,
+		Five = 5,
+		Six = 6,
+		Seven = 7,
+		Eight = 8,
+		Nine = 9,
+		Ten = 10,
+		Eleven = 11,
```
Comment (0x5bfa) [inline src/Files.Core/Data/Enums/GridViewSizeKind.cs:19]: Naming ok?

**Review by 0x5bfa — COMMENTED** [review]
Doesn' need Icon in DetailsView**Icon**SizeKind (all of them)?

File: src/Files.Core/Data/Enums/GridViewSizeKind.cs (hunk 9 above)
Comment (yaira2) [reply src/Files.Core/Data/Enums/GridViewSizeKind.cs:19]: Not really but I don't know what else to use.

Comment (0x5bfa) [reply src/Files.Core/Data/Enums/GridViewSizeKind.cs:19]: Right. I will add comments in another pr.

**Review by 0x5bfa — COMMENTED** [review]
I can't approve bc this is mine but https://github.com/files-community/Files/labels/approved

**Review by hishitetsu — APPROVED** [review]

**Pull request merged**